hs_discover/views.py

```diff
@@ -1,57 +1,224 @@
+                with DataOne standards. These standards do not contain an author field. Instead, the creator field
+                represents authors.
+        "contributor": list value, js will parse JSON as Array
+        "owner": list value, js will parse JSON as Array
+        "subject": list value, js will parse JSON as Array
+        "coverage_type": list point, period, ...
+        """
+        start = time.time()
+
+        sqs = SearchQuerySet().all()
+
+        if request.GET.get('filterbuilder'):
+            sqs = SearchQuerySet().facet('author')
+            authors = sqs.facet_counts()['fields']['author'][:self.filterlimit]
+            sqs = SearchQuerySet().facet('owner')
+            owners = sqs.facet_counts()['fields']['owner'][:self.filterlimit]
+            sqs = SearchQuerySet().facet('subject')
+            subjects = sqs.facet_counts()['fields']['subject'][:self.filterlimit]
+            sqs = SearchQuerySet().facet('contributor')
+            contributors = sqs.facet_counts()['fields']['contributor'][:self.filterlimit]
+            sqs = SearchQuerySet().facet('resource_type')
+            types = sqs.facet_counts()['fields']['resource_type'][:self.filterlimit]
+            sqs = SearchQuerySet().facet('availability')
+            availability = sqs.facet_counts()['fields']['availability'][:self.filterlimit]
+
+            return Response({
+                'time': (time.time() - start),
+                'filterdata': json.dumps([authors, owners, subjects, contributors, types, availability])
+            })
+
+        asc = '-1'
+        if request.GET.get('asc'):
+            asc = request.GET.get('asc')
+
+        sort = 'modified'
+        if request.GET.get('sort'):
+            sort = request.GET.get('sort')
+        sort = sort if asc == '1' else '-{}'.format(sort)
+
+        if request.GET.get('q'):
+            q = request.GET.get('q')
+            sqs = sqs.filter(content=q)  # .boost('keyword', 2.0)
 
-        sqs = SearchQuerySet()
+        try:
+            qs = request.query_params
+            filters = json.loads(qs.get('filter'))
+            # filter values expect lists, for example discoverapi/?filter={"owner":["Firstname Lastname"]}
+            if filters.get('author'):
+                sqs = sqs.filter(author__in=filters['author'])
+            if filters.get('owner'):
+                sqs = sqs.filter(owner__in=filters['owner'])
+            if filters.get('subject'):
+                sqs = sqs.filter(subject__in=filters['subject'])
+            if filters.get('contributor'):
+                sqs = sqs.filter(contributor__in=filters['contributor'])
+            if filters.get('type'):
+                sqs = sqs.filter(resource_type__in=list(filters['type']))
+            if filters.get('availability'):
+                sqs = sqs.filter(availability__in=filters['availability'])
+            if filters.get('geofilter'):
+                sqs = sqs.filter(north__range=[-90, 90])
```

Comment:
I know you explained this one to me, but I forgot what it is already.  Probably worth a code comment.

Reply:
Commit d41715b42